stdlib/public/Concurrency/Actor.cpp

```diff
@@ -676,6 +674,12 @@ class DefaultActorImpl : public HeapObject {
 
   friend class ProcessInlineJob;
   union {
+    // When the ProcessInlineJob storage is initialized, its metadata pointer
+    // will point to Job's metadata. When it isn't, the metadata pointer is
+    // NULL. Allow direct access to the metadata pointer so we can determine
+    // which state we're in.
+    Metadata *JobStorageMetadataPtr = nullptr;
```

Comment:
Why do this with a union? (Is it because of the `__ptrauth_objc_isa_pointer` on metadata? i.e. would `JobStorage.metadata` not work?)

Reply:
I'm afraid of C++ punishing me for trying to access a field of an uninitialized object. That fear may be unfounded.

Reply:
When the object _has_ been initialized, accessing `JobStorageMetadataPtr` will be UB, so it might punish you anyway (e.g. by assuming that it's always `nullptr` because accessing it when it wasn't is impossible).

Reply:
C allows you to access the common initial elements of two structs in a union, so I'm almost but not quite on a legal path here. If I wrap it in a `struct` then this should be good.